relayer/chains/mock/message_handlers.go

```diff
@@ -0,0 +1,90 @@
+package mock
+
+import (
+	"fmt"
+
+	chantypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	"github.com/cosmos/relayer/v2/relayer/ibc"
+	"github.com/cosmos/relayer/v2/relayer/provider"
+	"github.com/cosmos/relayer/v2/relayer/provider/cosmos"
+	"go.uber.org/zap"
+)
+
+type MsgHandlerParams struct {
+	mcp           *MockChainProcessor
+	PacketInfo    *chantypes.Packet
+	FoundMessages map[ibc.ChannelKey]map[string]map[uint64]provider.RelayerMessage
```

Comment:
map-ception

Reply:
Not a change request, just general advice:

I've found creating a type can help with readability and you can create methods off that type as well. 

E.g.

```go
type Messages map[ibc.ChannelKey]map[string]map[uint64]provider.RelayerMessage

func (m Messages) Retain(...) {
    // implementation
}

```

Note: `Messages` is probably not a great name. `MessageCache`?

If important enough, you can more easily unit test the new type. 

But, I will preach this to the ends of the earth, I never test private methods or types. They are tested through whatever public interface they support.

Reply:
will definitely be good for readability. will make that change.